### ISSUE TITLE
Node stopped with "Failed to write response" error

### ISSUE THREAD

**Issue by RiccardoM** [tendermint/tendermint#5475]
### Tendermint version
```
tendermint: 0.33.7
abci: 0.16.2
blockprotocol: 10
p2pprotocol: 7
```

### ABCI app
[Desmos v0.12.2](https://github.com/desmos-labs/desmos/tree/v0.12.2), which depends on [Cosmos v0.39.1](https://github.com/cosmos/cosmos-sdk/tree/v0.39.1).

### Environment
- **OS** (e.g. from /etc/os-release): Ubuntu 18.04

### What happened
Today, one of our chain nodes has randomly stopped with the following error: 

```
cosmovisor[5635]: E[2020-10-07|18:59:15.344] Failed to write response                     module=rpc-server protocol=websocket remote=161.81.171.147:61015 msg="RPCResponse
```

### Have you tried the latest version
No

### Logs
[Pastebin](https://pastebin.com/bYD1rpAc)

### Notes
We saw similar behaviour from explorer nodes when the RPC is requested frequently. This happened on Kava and Band before.

**Comment by melekes** [comment]
Did the node halt completely? Have you tried using `tendermint debug` https://medium.com/tendermint/rule-76-no-excuses-debug-like-a-champion-8ef42fc52786? Could you get us a stack trace after the node halts? I'm not 100% sure node halted because of the RPC.

Also, looks like we need to limit the number of characters in `msg="RPCResponse "` - it's too big!

**Comment by RiccardoM** [comment]
@melekes Yes, the node halted completely. We did not try using `tendermint debug` and we already re-started it. We will however do it next time.

**Comment by marbar3778** [comment]
@RiccardoM have you run into this issue again?

**Comment by RiccardoM** [comment]
> @RiccardoM have you run into this issue again?

Not recently, no

**Comment by RiccardoM** [comment]
@marbar3778 @melekes This happened again today. I was able to run `tendermint debug kill` and get the dump of the state. 

However, I noticed that the node did not halt completely. Instead, it continued working properly and it appears that only the RPC/gRPC stopped responding for some reason. I noticed this because I saw that our parser stopped working but the node was syncing normally looking at its status.

Here is the dump: 
[desmos-state.zip](https://github.com/tendermint/tendermint/files/6921243/desmos-state.zip)
